classes/class-p4-search.php

```diff
@@ -43,45 +45,195 @@ public function __construct( $search_query, $templates = [ 'search.twig', 'archi
 				$this->context = $context;
 			} else {
 				$this->context = Timber::get_context();
+				$this->handle_submit( $this->context );
 
+				if ( $this->search_query ) {
+					$this->all_posts    = $this->get_posts();
+					$this->current_page = get_query_var( 'paged' ) ? get_query_var( 'paged' ) : 1;
+					$this->posts        = array_slice( $this->all_posts, ( $this->current_page - 1 ) * self::POSTS_PER_PAGE, self::POSTS_PER_PAGE );
+				}
+				$this->set_context( $this->context );
+			}
+		}
+
+		/**
+		 * Handle form submit.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		public function handle_submit( &$context ) {
+			if ( 'GET' === $_SERVER['REQUEST_METHOD'] ) {
+
+				// Handle submitted sort options.
+				$selected_sort = filter_input( INPUT_GET, 'orderby', FILTER_SANITIZE_STRING );
+				if ( ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
+					$context['selected_sort'] = self::DEFAULT_SORT;
+				} else {
+					$context['selected_sort'] = $selected_sort;
+				}
+
+				// Handle submitted filter options.
+				if ( is_array( $_GET['f'] ) ) {
```

Comment:
Again I think we should pass the search query parameters in another way in the class constructor to eliminate superglobals use in the class. That way the class would be more easy to test the class and it's functions.